locales/en.json

```diff
@@ -97,15 +97,24 @@
     },
     "DropZone": {
```

Comment:
Would it make it easier to organize this as such:

```
    "DropZone": {
      "FileUpload": {
        "allowMultiple": {
          "file": {
            "overlayText": "Drop files to upload"
          },
          "image": {
            "overlayText": "Drop images to upload"
          }
        },
        "single": {
          "file": {
            "overlayText": "Drop file to upload"
          },
          "image": {
            "overlayText": "Drop image to upload"
          }
        }
      }
    }
```

then do something like this at the top of the file `DropZone`

```
  const translationNameSpace = 'Polaris.DropZone';
  const allowMultipleKey = allowMultiple ? 'allowMultiple' : 'single';
  const fileType = capitalize(type);

  const translationKey = `${translationNameSpace}.${allowMultipleKey}.${fileType}`;

  const overlayTextWithDefault =
    overlayText === undefined
      ? i18n.translate(`${translationKey}.overlayText`)
      : overlayText;
```

Reply:
I have no strong opinion about it, my goal was to preserve existing translations as much as I can.